### ISSUE TITLE
Using MSstatsTMT with a more complicated experimental design

### ISSUE THREAD

**Issue by twesleyb** [Vitek-Lab/MSstatsTMT#15]
Hi MSstats group,

I have struggled to  get my data into the MSstatsTMT pipeline. The reason for this is twofold:

1) My PSM report from `ProteomeDiscover` was saved as an `excel` document and thus its column names differ from what is expected by `MSstatsTMT`. 

The column names expected by MSstats have had all spaces and special characters replaced with `.`, e.g. 
`"Spectrum File" == "Spectrum.File"`. This is only a nuisance however, as I can replace these characters with `"."` myself

>  NOTE: the column 'Charge' is also required but not mentioned in the documentation. If there is no column 'Charge' MSstats stops with an error. 

2) MSstats expects that the `Spectrum.File` column from the PSM report should match `Run` in the user provided annotation file.

> Run : MS run ID. It should be the same as `Spectrum.File` info in `raw.pd`.

My problem is this: 
```
# There are 36 unique MS runs:
length(input.pd$Spectrum.File) ==  36

# There are 48 unique MS samples:
length(annotation.pd$Run) == 48
```

### Can MSstatsTMT handle my experimental design?

We performed 3 TMT experiments, in each analyzing 16-TMT labeled samples concatenated into a single `Mixture`.
This mixture was fractionated into 12 samples (`TechRepMixture`?). Thus, there were 12 LCMS injections for each TMT experiment.
Thereby `3 x 12 = 36` injections and `Spectrum Files`. However, there are `16 x 3 = 48` TMT samples.

The design looks like this:

|Sample                                        |Experiment |Channel |Treatment |Fraction |
|:---------------------------------------------|:----------|:-------|:---------|:--------|
|Abundance.F1.126.Sample.53395.Control4.Exp1   |Exp1       |126     |Control   |F4       |
|Abundance.F1.127C.Sample.53396.Control5.Exp1  |Exp1       |127C    |Control   |F5       |
|Abundance.F1.128C.Sample.53397.Control6.Exp1  |Exp1       |128C    |Control   |F6       |
|Abundance.F1.129C.Sample.53398.Control7.Exp1  |Exp1       |129C    |Control   |F7       |
|Abundance.F1.130C.Sample.53399.Control8.Exp1  |Exp1       |130C    |Control   |F8       |
|Abundance.F1.131C.Sample.53400.Control9.Exp1  |Exp1       |131C    |Control   |F9       |
|Abundance.F1.132C.Sample.53401.Control10.Exp1 |Exp1       |132C    |Control   |F10      |
|Abundance.F1.127N.Sample.53402.Mutant4.Exp1   |Exp1       |127N    |Mutant    |F4       |
|Abundance.F1.128N.Sample.53403.Mutant5.Exp1   |Exp1       |128N    |Mutant    |F5       |
|Abundance.F1.129N.Sample.53404.Mutant6.Exp1   |Exp1       |129N    |Mutant    |F6       |
|Abundance.F1.130N.Sample.53405.Mutant7.Exp1   |Exp1       |130N    |Mutant    |F7       |
|Abundance.F1.131N.Sample.53406.Mutant8.Exp1   |Exp1       |131N    |Mutant    |F8       |
|Abundance.F1.132N.Sample.53407.Mutant9.Exp1   |Exp1       |132N    |Mutant    |F9       |
|Abundance.F1.133N.Sample.53408.Mutant10.Exp1  |Exp1       |133N    |Mutant    |F10      |
|Abundance.F2.126.Sample.53409.Control4.Exp2   |Exp2       |126     |Control   |F4       |
|Abundance.F2.127C.Sample.53410.Control5.Exp2  |Exp2       |127C    |Control   |F5       |
|Abundance.F2.128C.Sample.53411.Control6.Exp2  |Exp2       |128C    |Control   |F6       |
|Abundance.F2.129C.Sample.53412.Control7.Exp2  |Exp2       |129C    |Control   |F7       |
|Abundance.F2.130C.Sample.53413.Control8.Exp2  |Exp2       |130C    |Control   |F8       |
|Abundance.F2.131C.Sample.53414.Control9.Exp2  |Exp2       |131C    |Control   |F9       |
|Abundance.F2.132C.Sample.53415.Control10.Exp2 |Exp2       |132C    |Control   |F10      |
|Abundance.F2.127N.Sample.53416.Mutant4.Exp2   |Exp2       |127N    |Mutant    |F4       |
|Abundance.F2.128N.Sample.53417.Mutant5.Exp2   |Exp2       |128N    |Mutant    |F5       |
|Abundance.F2.129N.Sample.53418.Mutant6.Exp2   |Exp2       |129N    |Mutant    |F6       |
|Abundance.F2.130N.Sample.53419.Mutant7.Exp2   |Exp2       |130N    |Mutant    |F7       |
|Abundance.F2.131N.Sample.53420.Mutant8.Exp2   |Exp2       |131N    |Mutant    |F8       |
|Abundance.F2.132N.Sample.53421.Mutant9.Exp2   |Exp2       |132N    |Mutant    |F9       |
|Abundance.F2.133N.Sample.53422.Mutant10.Exp2  |Exp2       |133N    |Mutant    |F10      |
|Abundance.F3.126.Sample.53423.Control4.Exp3   |Exp3       |126     |Control   |F4       |
|Abundance.F3.127C.Sample.53424.Control5.Exp3  |Exp3       |127C    |Control   |F5       |
|Abundance.F3.128C.Sample.53425.Control6.Exp3  |Exp3       |128C    |Control   |F6       |
|Abundance.F3.129C.Sample.53426.Control7.Exp3  |Exp3       |129C    |Control   |F7       |
|Abundance.F3.130C.Sample.53427.Control8.Exp3  |Exp3       |130C    |Control   |F8       |
|Abundance.F3.131C.Sample.53428.Control9.Exp3  |Exp3       |131C    |Control   |F9       |
|Abundance.F3.132C.Sample.53429.Control10.Exp3 |Exp3       |132C    |Control   |F10      |
|Abundance.F3.127N.Sample.53430.Mutant4.Exp3   |Exp3       |127N    |Mutant    |F4       |
|Abundance.F3.128N.Sample.53431.Mutant5.Exp3   |Exp3       |128N    |Mutant    |F5       |
|Abundance.F3.129N.Sample.53432.Mutant6.Exp3   |Exp3       |129N    |Mutant    |F6       |
|Abundance.F3.130N.Sample.53433.Mutant7.Exp3   |Exp3       |130N    |Mutant    |F7       |
|Abundance.F3.131N.Sample.53434.Mutant8.Exp3   |Exp3       |131N    |Mutant    |F8       |
|Abundance.F3.132N.Sample.53435.Mutant9.Exp3   |Exp3       |132N    |Mutant    |F9       |
|Abundance.F3.133N.Sample.53436.Mutant10.Exp3  |Exp3       |133N    |Mutant    |F10      |

In each experiment, two `Subjects` (mice) were analyzed. A 'Control' (WT) and 'Mutant' mouse from which 7 biological `Fractions` were prepared (`F2-F10`). 
> __NOTE:__ Here `Fraction` is a subcellular fraction prepared by differential centrifugation, and not the same as the  12 fractions prepared from a single TMT mixture for each experiment.

So there are only 36 possible `Spectrum.Files` and 48 TMT `Runs` as I understand it.
I would like to make comparisons at two levels:
1) Intrafraction comparisons: `~ 0 + interaction(Fraction,Treatment)` e.g. `'F4.Control vs F4.Mutant'` and `'F6.Control vs F6.Mutant'`.
2) `'Control vs Mutant'` adjusted for fraction differences: ` ~ Fraction + Treatment`

I have inspected the source code and can try to fit the protein level calls to lmer myself. However, I a not quite sure I understand all of the design formula om terms of Mixture Group Subject and TechRepMixture plus my additional Factor `Fraction`.

```
## calls to lmerTest in MSstatsTMT::linearModel.functions.R

# model 1
lmer(Abundance ~ 1 + (1|Mixture) + (1|Mixture:TechRepMixture) + Group + (1|Subject:Group:Mixture), data = data)
 
# model 2
lmer(Abundance ~ 1 + (1|Run) + Group + (1|Subject:Group), data)

# model 3
lmer(Abundance ~ 1 + (1|Mixture) + (1|Mixture:TechRepMixture) + Group, data = data)

# model 4
lmer(Abundance ~ 1 + (1|Run) + Group, data = data)

# model 5
lm(Abundance ~ 1 + Group, data = data)
```

#### TLDR
In short, how can I pass my experimental design to MSstatsTMT?

I know this is a lot. Any help is appreciated.

-tyler

**Comment by twesleyb** [comment]
Whoops, I just found your recent paper:
https://www.mcponline.org/content/mcprot/early/2020/07/17/mcp.RA120.002105.full.pdf

This may answer all my questions.
